lib/services/meta_task_db_service.dart

```diff
@@ -1,108 +1,98 @@
-    log.infoObj({'method': 'getAllTasks'});
-    try {
-      List<MetaTask> metaTaskList = [];
-      QuerySnapshot querySnapshot = await taskCollection.get();
-      if (querySnapshot.docs.isNotEmpty) {
-        for (var doc in querySnapshot.docs.toList()) {
-          metaTaskList.add(new MetaTask.fromData(doc.data()));
-        }
-      }
-      log.infoObj({'method': 'getAllTasks - success', 'metaTaskList': metaTaskList});
-      return {'metaTaskList': metaTaskList};
-    } catch (error) {
-      log.errorObj({'method': 'getAllTasks - error', 'error': error.toString()});
-      return {'error': error};
-    }
-  }
-
-  // Retrieving a task by their difficulty level and the title
-  Future<Map<String, dynamic>> getTaskByTitleAndDifficulty(String title, String difficulty) async {
-    log.infoObj({
-      'method': 'getTaskByTitleAndDifficulty',
-      'title': title,
-      'difficulty': difficulty
-    });
-    try {
-      String id = generateTaskId(title + difficulty);
-      DocumentSnapshot task = await taskCollection.doc(id).get();
-
-      // User does not exist
-      if (!task.exists) {
-        String message = 'Task does not exist';
-        log.errorObj({'method': 'getTaskByTitleAndDifficulty', 'error': message});
-        return {'error': message};
-      }
-      MetaTask metaTask = new MetaTask.fromData(task.data());
-      log.successObj({
-        'method': 'getTaskByTitleAndDifficulty - success',
-        'metaTask': metaTask
-      });
-      return {'metaTask': metaTask};
-    } catch (error) {
-      log.errorObj({
-        'method': 'getTaskByTitleAndDifficulty - error',
-        'error': error.toString()
-      });
-      return {'error': error};
-    }
-  }
-
-  // Hashing the task title as unique id to store inside db
-  String generateTaskId(String titleDifficulty) {
-    return sha256.convert(utf8.encode(titleDifficulty)).toString();
-  }
-}
+import 'package:cloud_firestore/cloud_firestore.dart';
+import 'package:curiosity_flutter/models/meta_task.dart';
+import 'package:curiosity_flutter/services/log_service.dart';
+
+class MetaTaskDbServices {
+  final String META_TASK_DB_NAME = 'meta-task-dev';
+  final LogService log = new LogService();
+  CollectionReference taskCollection;
+
+  MetaTaskDbServices() {
+    taskCollection = FirebaseFirestore.instance.collection(META_TASK_DB_NAME);
+  }
+
+  // Creating new task, expect passed in data to contain
+  // 1. title - title of the task
+  // 2. difficulty - difficulty | easy, intermediate, hard
+  // 3. description - description of the task
+  // 4. proofDescription - description on proof of task completion
+  Future<Map<String, dynamic>> addMetaTask(Map<String, String> data) async {
+    log.infoObj({'method': 'addMetaTask', 'data': data});
+    try {
+      // id created by hashing the title and difficulty of a task
+      DocumentSnapshot documents = await taskCollection.doc(data['difficulty']).get();
+      int count = documents.get("count");
+

```

Comment:
Should you account for cases where the users attempt to add two tasks of the same difficulty and same task name? We should not allow that.

Reply:
Only an admin should be adding tasks and having the same difficulty/name won't cause conflicts or break anything. We should just assume that whoever is adding tasks doesn't do this, otherwise we would have to query every task we store each time we want to add a new task.